src/providers/virtual/qgsvirtuallayerfeatureiterator.cpp

```diff
@@ -105,6 +106,13 @@ QgsVirtualLayerFeatureIterator::QgsVirtualLayerFeatureIterator( QgsVirtualLayerF
         wheres << values;
       }
     }
+    else
+    {
+      if ( request.filterType() == QgsFeatureRequest::FilterFid )
+      {
+        offset = QStringLiteral( " LIMIT 1 OFFSET %1" ).arg( request.filterFid() );
```

Comment:
I think there is a problem with negative ids here. If you pass, say, -23, it will return the first row rather than no row ... because OFFSET -23 == OFFSET 0

Reply:
Would this ever happen though? Do virtual layers use negative ids for anything?